### DISCUSSION TITLE
SimpleTcpClient disconnects right after connection

### DISCUSSION THREAD

**Discussion by Dried09** [jchristn/SuperSimpleTcp#113] [Q&A]
HI everyone,

Using SimpleTcpClient with Unity, server is console app on .net 6 with SimpleTcpServer and faced with issue that Unity-side client connection is disconnecting right after connection. It seemed obvious to use some keepalive settings but I had no luck, all the combinations of these settings led to the same result.

Logs from Unity:
![image](https://user-images.githubusercontent.com/24873164/153481579-88bce572-20c3-470d-8ccf-3b8ea22c3129.png)
Logs from server app:
![image](https://user-images.githubusercontent.com/24873164/153481661-f3d4499d-1a0e-4d81-99f0-24ceb2a49e14.png)

Client connection code:
` public void ConnectToServer(string serverIp, int port)
        {
            Telepathy.Log.Info = Debug.Log;
            Telepathy.Log.Warning = Debug.LogWarning;
            Telepathy.Log.Error = Debug.LogError;

            client = new SimpleTcpClient(serverIp, port);

            client.Events.Connected += ServerConnected;
            client.Events.Disconnected += ServerDisconnected;
            client.Events.DataReceived += DataReceived;

            client.Connect();
        } `
 Server app code:
 ` public void StartTransport()
        {
            Server = new SimpleTcpServer($"127.0.0.1:9000");

            Server.Events.ClientConnected += ClientConnected;
            Server.Events.ClientDisconnected += ClientDisconnected;
            Server.Events.DataReceived += DataReceived;

            Server.Start();
        } `
        
 Looking forward for some help with this.
 Thanks in advance.

**Comment by pure-hate** [comment]
There is no code after connection

**Reply by Alywan** [reply]
We're having a similar problem.
Randomly disconnects after 10-20-30 etc seconds.
If we disable KeepAlive, it works, but we really need to have KeepAlive, so that we can force reconnect once it detects disconnection.

**Reply by jchristn** [reply]
Hi @Alywan what framework/runtime and operating system are you using?  Can you provide code to reproduce?

**Reply by pure-hate** [reply]
try declaring a separate static public class SimpleTcpClient, when the TryToConnect function finishes its execution, the "client" variable is destroyed
